lisa/util/parallel.py

```diff
@@ -108,6 +109,10 @@ def wait_worker(self) -> bool:
             self._future_task_map.pop(future)
         return len(self._futures) > 0
 
+    def wait_for_all_workers(self) -> None:
```

Comment:
It's not a good way to wait all workers. Use a variable at line 100, replace the hardcoded `FIRST_COMPLETED`.

`done_futures, _ = wait(self._futures[:], return_when=FIRST_COMPLETED)`

Reply:
how do you like something like this:
```
    def wait_worker(self, return_condition: str = FIRST_COMPLETED) -> bool:
        """
        Return:
            True, if there is running worker.
        """

        done_futures, _ = wait(self._futures[:], return_when=return_condition)
        for future in done_futures:
            # join exceptions of subthreads to main thread
            result = future.result()
            # removed finished threads
            self._futures.remove(future)
            # exception will throw at this point
            self._callback(result)
            self._future_task_map[future].close()
            self._future_task_map.pop(future)
        return len(self._futures) > 0

    def wait_for_all_workers(self) -> None:
        remaining_workers = self.wait_worker(return_condition=ALL_COMPLETED)
        assert_that(remaining_workers).is_zero()

```

Reply:
Yes, there would be two line changes only.